### PR TITLE
Fix EZP-24967: Saved RichText content has auto-generated YUI ids

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-24967

## Description
YUI generates ids. We avoid using it for xml parsing and we now remove them along with any ids (as we don't need them).

## Tests
Manual and unit test

### CODE REVIEW

**Comment by yannickroger** [comment]
ping @dpobel @mhyndle @StephaneDiot

File: Resources/public/js/views/fields/ez-richtext-editview.js [hunk 1]
```diff
@@ -293,15 +293,29 @@ YUI.add('ez-richtext-editview', function (Y) {
...
-                section = Y.Node.create(data);
+                root, i, element, list, section,
+                doc = document.createDocumentFragment();
+
+            root = document.createElement('div');
+            doc.appendChild(root);
+            root.innerHTML = data;
+            list = root.querySelectorAll('[id]');
+
+            if (list.length) {
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-richtext-editview.js]: not needed

File: Resources/public/js/views/fields/ez-richtext-editview.js [hunk 2]
```diff
@@ -293,15 +293,29 @@ YUI.add('ez-richtext-editview', function (Y) {
...
+                for (i = 0; i != list.length; ++i) {
+                    element = list[i];
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-richtext-editview.js]: you don't really need a local variable, `list[i].removeAttribute("id");` would work well here.

File: Resources/public/js/views/fields/ez-richtext-editview.js [hunk 3]
```diff
@@ -293,15 +293,29 @@ YUI.add('ez-richtext-editview', function (Y) {
...
+                    element.removeAttribute("id");
+                }
+            }
 
-            if ( section ) {
-                Y.Object.each(ROOT_SECTION_ATTRIBUTES, function (value, key) {
-                    section.removeAttribute(key);
+            section = root.firstElementChild;
```
Comment (dpobel) [inline Resources/public/js/views/fields/ez-richtext-editview.js]: I would write: `section = root.querySelector('section');` which adds a bit more information when reading the code.

File: Tests/js/views/fields/assets/ez-richtext-editview-tests.js [hunk 4]
```diff
@@ -309,6 +313,29 @@ YUI.add('ez-richtext-editview-tests', function (Y) {
                 "The xml property of the fieldValue should come from the editor"
             );
         },
+
+        "Should remove identifiers": function () {
```
Comment (dpobel) [inline Tests/js/views/fields/assets/ez-richtext-editview-tests.js]: nitpick: "Should remove id attributes"

**Comment by dpobel** [comment]
besides the comment, +1

**Comment by mhyndle** [comment]
+1

**Pull request merged**